### DISCUSSION TITLE
Unwanted control change event after disposing of Output Device

### DISCUSSION THREAD

**Discussion by TizianoBarattin** [melanchall/drywetmidi#271] [Q&A] [answered]
Hi everyone.
I have some problems with Output midi device. 
After disposing of it for sending a simple control change event, I see with a midi monitor application that there are some unwanted control change (one cc 121 and one cc 123 with value 0 for each of 16 channels).
With the breakpoint i see that it happens when I get out of "using" call. I tried also not to call the Dispose method after sending the message and in this case i see ccs passing when i call that method.
What am I doing wrong?
Here the code I use

using (var outputDevice = OutputDevice.GetByName("name"))
                    {
                        
                    
                        ControlChangeEvent message = new ControlChangeEvent((SevenBitNumber)10, (SevenBitNumber)10) 
                        {
                            Channel = (FourBitNumber)1    
                        };
                    
                        outputDevice.SendEvent(message);  
                    }

**Comment by melanchall** [comment, marked as answer]
Hi,

It's definitely how your device works. DryWetMIDI doesn't send any events on disposing.

When disposing happens, DWM just calls system API to close/release a MIDI device. So what you see is how your device reacts on such releasing.

Please look into the device manual to learn what's going on or ask a question on forum/tech support of the device vendor.

**Comment by TizianoBarattin** [comment]
I had this problem with two different devices. One is a Focusrite 18i20 2nd gen and the other is a virtual loop midi port created with "loopMidi". I don't know, maybe it is a problem of my pc. When I will have another pc available I will try with that.

**Reply by melanchall** [reply]
> I had this problem with two different devices

It's strange but again, DWM can't send all those events. The library is open-source one, so you can take a look into sources and be sure that I'm saying truth :-)
